### PR TITLE
Hexagon CI coverage

### PR DESCRIPTION
**Pull request by nickdesaulniers**
We don't want to merge this until https://gitlab.com/Linaro/tuxsuite/-/issues/107 is resolved, otherwise builders will fail.

### CODE REVIEW

**Review by nathanchance — APPROVED** [review]
Looks good to me. I tested locally with both self-compiled clang as well as the `tuxmake` Docker images, everything built fine.

File: generator.yml [hunk 1]
```diff
@@ -48,6 +51,7 @@ architectures:
 triples:
   - &arm-triple         {CROSS_COMPILE: arm-linux-gnueabi-,    ARCH: *arm-arch}
   - &arm64-triple       {CROSS_COMPILE: aarch64-linux-gnu-,    ARCH: *arm64-arch}
+  - &hexagon-triple     {CROSS_COMPILE: hexagon-linux-gnu-,    ARCH: *hexagon-arch}
```
Comment (androm3da) [inline generator.yml:54]: It may not matter much at this point but I would recommend `hexagon-unknown-linux-musl-`.

**Comment by nathanchance** [comment]
These changes are going into mainline: https://lore.kernel.org/r/a81e116ba3d64aab8d379703bedcf032@codeaurora.org/

Once TuxSuite gains support for Hexagon, we should just build mainline and next I think.

File: generator.yml (hunk 1 above)
Comment (nickdesaulniers) [reply generator.yml]: done

**Comment by nickdesaulniers** [comment]
updated to refer to mainline+next

**Review by nathanchance — APPROVED** [review]
LGTM once TuxSuite supports Hexagon.

**Comment by nickdesaulniers** [comment]
x-ref: https://gitlab.com/Linaro/tuxsuite/-/issues/107

File: generator.yml [hunk 2]
```diff
@@ -48,6 +49,7 @@ architectures:
 triples:
   - &arm-triple         {CROSS_COMPILE: arm-linux-gnueabi-,    ARCH: *arm-arch}
   - &arm64-triple       {CROSS_COMPILE: aarch64-linux-gnu-,    ARCH: *arm64-arch}
+  - &hexagon-triple     {CROSS_COMPILE: hexagon-linux-musl-,   ARCH: *hexagon-arch}
```
Comment (terceiro) [inline generator.yml:52]: I don't really understand the setup here, so I have little input to provide on the rest. But this bit was useful to me, thanks.

**Comment by nathanchance** [comment]
Sounds like this should be resolved on the TuxSuite side of things according to https://gitlab.com/Linaro/tuxsuite/-/issues/107? Rebase then merge? Don't forget to regen the TuxSuite and workflow files because of #134 and #137.

**Comment by nathanchance** [comment]
I rebased this myself, @nickdesaulniers look good still?

**Comment by nickdesaulniers** [comment]
yes thanks @nathanchance ! and @terceiro and @vishalbhoj !

**Comment by nickdesaulniers** [comment]
and a big thanks to @androm3da and Sid!

**Comment by androm3da** [comment]
Thanks to @nickdesaulniers and @terceiro and team for including hexagon!  I am subscribed to ClangBuiltLinux but if I miss out, please notify me about any toolchain or kernel regressions and we will address them.

**Pull request merged**